modules/apps/frontend-js/frontend-js-item-selector-web/src/main/resources/META-INF/resources/item_selector/ItemSelectorModal.tsx

```diff
@@ -150,6 +158,8 @@ function ItemSelectorModal<T extends Record<string, any>>({
 							className="btn-cancel"
 							displayType="secondary"
 							onClick={() => {
+								setSelectedItems([]);
```

Comment:
I have removed this was clearing the selected items, losing track on a second modal opening. Can you, please, check that it is not happening again?

Reply:
BTW, it is probably working because of the `useEffect(...)` on line 87 so, maybe, this `setSelectedItems()` is not needed at all.

Reply:
I think this `setSelectedItems` makes sense because we want to clear the selected state of the items in the case where the modal is closed. I think it would be weird to close the modal, reopen it, and then see the same items selected.

The useEffect looks like it is there to sync the `externalItems` to the selected state, since those are really the "default" selected values.

Does that make sense?


IMO if you "cancel" or "close" a modal, you shouldn't preserve any state that was selected in that modal(unless it was already submitted)

Reply:
Yes that's correct. Cancelling should discard any changes you had made in the modal. When we reopen it, it should return to the showing the no items or the previosly confirmed items.

Reply:
It is currently showing the previously confirmed ones because of the `useEffect()`, and it the `externalItems` are empty, the it will be "cleared". That's why I'm saying `setSelectedItems([]);` may not be necessary.